File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []

        if not root:
            return res

        queue = deque([root])

        while queue:
            level = []

            for _ in range(len(queue)):
                cur = queue.popleft()

                level.append(cur.val)

                if cur.left:
                    queue.append(cur.left)
                
                if cur.right:
                    queue.append(cur.right)
                
            res.append(level)
        
        return res

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    
    if root == nil {
        return [][]int{}
    }
    res := make([][]int, 0)

    queue := make([]*TreeNode, 0)
    queue = append(queue, root)

    for len(queue) > 0 {
        level := make([]int, 0)

        size := len(queue)

        for i := 0; i < size; i++ {
            cur := queue[0]
            queue = queue[1:]

            level = append(level, cur.Val)

            if cur.Left != nil {
                queue = append(queue, cur.Left)
            }

            if cur.Right != nil {
                queue = append(queue, cur.Right)
            }
        }

        res = append(res, level)
    }

    return res
}